configure.ac

```diff
@@ -0,0 +1,27 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([mailio], [0.14.1], [contact@alepho.com])
+LT_INIT
+AC_CONFIG_SRCDIR([src/codec.cpp])
+AC_CONFIG_HEADERS([config.h])
+
```

Comment:
Adding something like
```
AC_CONFIG_AUX_DIR([build/scripts])
```
will put executable scripts into the separate directory. Is there a similar option for the generated directories `autom4t.cache` and `m4`, so they do not shuffle with the rest?

Reply:
In configure.ac change ```AC_CONFIG_MACRO_DIRS([m4])``` to ```AC_CONFIG_MACRO_DIRS([build/m4])```

In each Makefile.am change ```ACLOCAL_AMFLAGS = -I m4``` to ```ACLOCAL_AMFLAGS = -I build/m4```

Create a file in the root directory called .autom4te.cfg:
```## ------------------ ##
## User Preferences.  ##
## ------------------ ##

begin-language: "Autoconf"
args: --cache=build/autom4te.cache/
end-language: "Autoconf"
```

Reply:
Thanks, it works. Did you manage to compile examples? I am investigating why the linker complains that Boost.Random is missing, since all ´-l´ switches are there. The library itself is compiled - both static and dynamic ones.

Reply:
I can compile the examples by manually specifying -lboost_random in examples/Makefile.am
```AM_LDFLAGS = -lpthread -L../src $(BOOST_SYSTEM_LIBS) $(BOOST_REGEX_LIBS) $(BOOST_DATE_TIME_LIBS) -lboost_random $(SSL_LIBS) $(CRYPTO_LIBS) -lmailio``` 

For some reason there is no m4 macro for Boost.Random, so ```./configure``` can't find it.

Reply:
It appears that `-lmailio` in `examples/Makefile.am` should be placed before Boost libraries:
```
LDADD = -lpthread -L../src -lmailio $(BOOST_REGEX_LIBS) $(BOOST_DATE_TIME_LIBS) -lboost_random $(SSL_LIBS) $(CRYPTO_LIBS) $(BOOST_SYSTEM_LIBS)
```
Otherwise, linker cannot find Boost.Random.